api/app/controllers/spree/api/address_books_controller.rb

```diff
@@ -2,32 +2,41 @@ module Spree
   module Api
     class AddressBooksController < Spree::Api::BaseController
       # Note: the AddressBook is the resource to think about here, not individual addresses
+      before_filter :authorize_user_address
+      before_filter :load_user_addresses
 
       def show
-        render_address_book
+        render :show, status: :ok
       end
 
       def update
         address_params = address_book_params
         default_flag = address_params.delete(:default)
-        address = current_api_user.save_in_address_book(address_params, default_flag)
+        address = address_book_user.save_in_address_book(address_params, default_flag)
         if address.valid?
-          render_address_book
+          render :show, status: :ok
         else
           invalid_resource!(address)
         end
       end
 
       def destroy
-        current_api_user.remove_from_address_book(params[:address_id])
-        render_address_book
+        address_book_user.remove_from_address_book(params[:address_id])
+        render :show, status: :ok
       end
 
       private
 
-      def render_address_book
-        @user_addresses = current_api_user.user_addresses
-        render :show, status: :ok
+      def authorize_user_address
+        authorize! :manage, UserAddress.new(user_id: address_book_user.id)
```

Comment:
:+1: nice. I think that looks like the best option.

Since we don't have an "AddressBook" model, these things are all really a property of the user, so to me it makes sense to make the auth on the user like that.

Maybe another option would be to just assume that "update User" should allow updating the user's addresses (in the same way that we don't have separate permissions for updating a user's first name vs their last name vs their email), but addresses seem worth breaking out permissions-wise so I think your branch is the better choice of those.